python_modules/libraries/dagster-airbyte/dagster_airbyte/resources.py

```diff
@@ -77,7 +77,9 @@ def make_request(self, endpoint: str, data: Optional[Dict[str, Any]]):
                     timeout=15,
                 )
                 response.raise_for_status()
-                return response.json()
+                if response.status_code != 204:
```

Comment:
In the case that the response is empty, is there benefit to returning the actual response object?

Right now, the return type of this function is Dict[str, Any], because we always expect to be returning some json data, and sometimes returning a raw response object would require changing the type signature a fair bit. Instead, we could consider slightly modifying the type signature of this function to Optional[Dict[str, Any]], and returning None in the case that we get a 204 response.

Reply:
Thanks for your review, @OwenKephart.

> Instead, we could consider slightly modifying the type signature of this function to Optional[Dict[str, Any]], and returning None in the case that we get a 204 response.

Indeed, that suggestion sounds good/simple. I've pushed [2c08fd9](https://github.com/dagster-io/dagster/pull/7209/commits/2c08fd9fb73bcc6412351cd81a2c8064be622c79) that incorporates this suggestion.